From c6fe26b74ab95a719b46526c7f3b1dfd9689626a Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Tue, 22 Jan 2013 18:40:25 +0800
Subject: [PATCH 048/231] ENGR00241251 imx6_sabreauto: workaround touch screen
 can not wake up issue.

Workaround this issue by not suspend, and this will be check by driver,
if can not request the wake up gpio, it will not do the suspend & resume.

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_sabreauto.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/mach-mx6/board-mx6q_sabreauto.c b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
index acd42ba..dcc654a 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabreauto.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
@@ -1704,6 +1704,14 @@ static void __init mx6_board_init(void)
 	imx6q_add_perfmon(0);
 	imx6q_add_perfmon(1);
 	imx6q_add_perfmon(2);
+
+	/* acquire this irq is want to let touch driver failed to
+	 * request this irq gpio, and the driver will disable suspend
+	 * funtion, to workaround sabreauto board have isssue on gpio
+	 * of touch use. */
+	gpio_request(SABREAUTO_CAP_TCH_INT, "touch irq");
+	gpio_direction_input(SABREAUTO_CAP_TCH_INT);
+
 }
 
 extern void __iomem *twd_base;
-- 
1.7.10.4

